pandas/core/arrays/string_.py

```diff
@@ -352,6 +352,9 @@ def _from_sequence(cls, scalars, *, dtype: Dtype | None = None, copy: bool = Fal
             result[na_values] = libmissing.NA
 
         else:
+            if hasattr(scalars, "type"):
```

Comment:
Why can't we do an actual isinstance check?

If we have some helper method `is_pyarrow_array` like the following, this should be quite easy? 

```
def is_pyarrow_array(obj):
    if _pyarrow_installed:
        return isinstance(obj, pa.Array)
    return False
```

with variables `_pyarrow_installed` and `pa` filled in once on first import.

Reply:
PR that does this -> https://github.com/pandas-dev/pandas/pull/52830